man/fi_rpc.3.md

```diff
@@ -0,0 +1,284 @@
+
+ssize_t fi_rpc_resp(struct fid_ep *ep, const void *buf, size_t len,
+	void *desc, uint64_t rpc_id, void *context);
+
+ssize_t fi_rpc_respv(struct fid_ep *ep, const struct iovec *iov, void **desc,
+	size_t count, uint64_t rpc_id, void *context);
+
+ssize_t fi_rpc_respmsg(struct fid_ep *ep, const struct fi_msg_rpc_resp *msg,
+	uint64_t flags);
+```
+
+# ARGUMENTS
+
+*ep*
+: Fabric endpoint on which to send RPC request or response.
+
+*buf*
+: Data buffer to send RPC request or response.
+
+*resp_buf*
+: Data buffer to receive RPC response.  The content of this buffer is only
+  valid if the operation completes successfully.
+
+*len*
+: Length of data buffer to send RPC request or response, specified in bytes.  Valid
+  transfers are from 0 bytes up to the endpoint's max_msg_size.
+
+*resp_len*
+: Length of data buffer to receive RPC response, specified in bytes.  Valid
+  transfers are from 0 bytes up to the endpoint's max_msg_size.
+
+*iov*
+: Vectored data buffer.
+
+*resp_iov*
+: Vectored data buffer to receive RPC response.
+
+*count*
+: Count of vectored data entries.
+
+*resp_count*
+: Count of vectored data entries to receive RPC response.
+
+*desc*
+: Descriptor associated with the data buffer.  See [`fi_mr`(3)](fi_mr.3.html).
+
+*resp_desc*
+: Descriptor associated with the data buffer to receive RPC response.  See
+  [`fi_mr`(3)](fi_mr.3.html).
+
+*dest_addr*
+: Destination address for connectionless transfers.  Ignored for
+  connected endpoints.
+
+*msg*
+: Message descriptor for send RPC request or response.
+
+*flags*
+: Additional flags to apply for the operation.
+
+*timeout*
+: Timeout value for the RPC request.  The value is specified in microseconds.
+  A value of 0 indicates that the operation should not timeout.  A value of
+  -1 indicates that the operation should use the default timeout value
+  configured for the endpoint.
+
+*rpc_id*
+: RPC identifier for the request.  This value is used to match the request
+  with the response.
+
+*context*
+: User specified pointer to associate with the operation.  This parameter is
+  ignored if the operation will not generate a successful completion, unless
+  an op flag specifies the context parameter be used for required input.
+
+# DESCRIPTION
+
+RPC style data transfer involves the client sending a request to the server
+and the server sending back a response. The buffer for holding the response
+is supplied with the request.  The content of the request and response is
+defined by the application.
+
+An RPC request is delivered into a data buffer posted by message receive
+functions -- fi_recv, fi_recvv, fi_recvmsg.  The completion of the receive
+operation has FI_RPC set in the `flags` field.  The full completion entry
+is defined as `struct fi_cq_rpc_entry`.  The server should set the CQ format
+to FI_CQ_FORMAT_RPC in order to get the necessary information for correct
+handling of the RPC request.
+
+The server will receive the RPC ID as part of the completion entry.  The RPC
```

Comment:
'The RPC ID associated with the request is reported as part of the completion entry.'